unit_tests/test_s3_remote_uploader.py

```diff
@@ -0,0 +1,143 @@
+import os
+import pathlib
+import tempfile
+import time
+import subprocess
+import boto3
+import pytest
+import logging
+import secrets
+import tarfile
+
+from sdcm.utils.s3_remote_uploader import upload_remote_files_directly_to_s3
+from sdcm.utils.common import S3Storage
+
+# Constants for the test
+TEST_S3_BUCKET = S3Storage.bucket_name
+TEST_S3_KEY_PREFIX = os.environ.get("TEST_S3_KEY_PREFIX", "integration-test/")
+SSH_IMAGE = "linuxserver/openssh-server:latest"
+SSH_USER = "sct-user"  # linuxserver/openssh-server default user
+SSH_PORT = 2222
+
+
+@pytest.fixture(scope="module")
+def ssh_docker_server(tmp_path_factory):
+    """
+    Pytest fixture to start an SSH server in a Docker container for testing.
+    Generates an SSH key pair, starts the container, waits for SSH to be available,
+    and yields SSH connection info, container name, and temp path. Cleans up the container after use.
+    """
+    tmp_path = tmp_path_factory.mktemp("ssh_server")
+    ssh_key_path = tmp_path / "id_ed25519"
+    ssh_pub_path = tmp_path / "id_ed25519.pub"
+    # Generate SSH key pair
+    subprocess.run([
+        "ssh-keygen", "-t", "ed25519", "-f", str(ssh_key_path), "-N", ""
+    ], check=True)
+    # Start SSH server in Docker
+    container_name = f"test-ssh-server-{os.getpid()}"
+    cmd = [
+        "docker", "run", "-d",
+        "--name", container_name,
+        "-p", f"{SSH_PORT}",
+        "-e", f"USER_NAME={SSH_USER}",
+        "-e", f"PUBLIC_KEY={pathlib.Path(ssh_pub_path).read_text().strip()}",
+        SSH_IMAGE
+    ]
+    # Print docker run output for debugging
+    logging.info('Running docker run command: %s', ' '.join(cmd))
+    docker_run_result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=False)
+    logging.info('docker run stdout: %s', docker_run_result.stdout.decode())
+    logging.info('docker run stderr: %s', docker_run_result.stderr.decode())
+    if docker_run_result.returncode != 0:
+        pytest.skip(f"Docker run failed: {docker_run_result.stderr.decode()}")
+
+    # Get the container's IP address on the docker bridge network
+    inspect_cmd = [
+        "docker", "inspect", "-f",
+        "{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}",
+        container_name
+    ]
+    docker_address = subprocess.run(inspect_cmd, check=True, stdout=subprocess.PIPE).stdout.decode().strip()
+    # Wait for SSH to be up
+    for _ in range(20):
+        try:
+            os.environ.pop("SSH_AUTH_SOCK", None)  # Ensure no SSH agent interference
```

Comment:
how about using monkeypatch instead of accessing os.environ directly?

Reply:
Sorry, I do not follow,

I meant using https://docs.pytest.org/en/stable/how-to/monkeypatch.html#monkeypatching-environment-variables, which should be isolated to not affect the rest of the tests. I know we are currently using os.environ directly in tests (potential improvement for later), but I would not introduce new usages if the monkeypatch will work here